challenge_1/rust/deg4uss3r/challenge_1/src/lib.rs

```diff
@@ -0,0 +1,14 @@
+pub fn string_reverse (mut in_str: std::string::String) -> std::string::String {
+
+    let mut out = "".to_string();
+    let in_string = in_str.len();
+    let mut i: usize = 0;
+
+    while i < in_string {
+        let out_temp = in_str.pop().unwrap().to_string();
+        out += &out_temp;
+        i+=1;
+    }
```

Comment:
That's one way to do this (and explains the mutable ownership parameter), but not the most idiomatic. Strings can coerce to double-ended iterators very easily, so `in.chars().rev().collect()` will split, tweak, and rejoin the input.